src/Symfony/Component/PropertyInfo/Extractor/PhpDocExtractor.php

```diff
@@ -227,9 +233,30 @@ private function getDocBlockFromMethod(string $class, string $ucFirstProperty, i
         }
 
         try {
-            return [$this->docBlockFactory->create($reflectionMethod, $this->contextFactory->createFromReflector($reflectionMethod)), $prefix];
+            return [$this->docBlockFactory->create($reflectionMethod, $this->createFromReflector($reflectionMethod)), $prefix];
         } catch (\InvalidArgumentException $e) {
             return null;
         }
     }
+
+    /**
+     * static cache around $this->contextFactory->createFromReflector()
+     * prevents a lot of redundant calls to createForNamespace.
+     */
+    private function createFromReflector(\Reflector $reflector): Context
```

Comment:
instead of `Reflector`, it'd suggest `@param \ReflectionClass|\ReflectionMethod $reflector`